py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py

```diff
@@ -195,6 +198,44 @@ def __init__(self, name=None, namespace=None,
 
     self.kfctl_path = os.path.join(self.src_dir, "bootstrap/bin/kfctl")
 
+    # Fetch the main repo from Prow environment.
+    self.main_repo = argo_build_util.get_repo_from_prow_env()
+
+    # extra_repos is a list of comma separated repo names with commits,
+    # in the format <repo_owner>/<repo_name>@<commit>,
+    # e.g. "kubeflow/tf-operator@12345,kubeflow/manifests@23456".
+    # This will be used to override the default repo branches.
+    self.extra_repos = []
+    if extra_repos:
+      self.extra_repos = extra_repos.split(',')
+
+  def _build_repos_str(self):
```

Comment:
Can we add a unittest?

Also how about making this a routine in kubeflow/testing so that it can be more easily reused by different tests?

Reply:
https://github.com/kubeflow/testing/pull/511